### PR TITLE
[1.x] Select Component: Add. Limit, Disabled and Image Option

### PR DESCRIPTION
**Pull request by hitaloramon**
### What:

- [ ] Bug Fix
- [x] Enhancements
- [ ] New Feature

### Description:

This PR aims to bring the following improvements to the select component:

1. Possibility to disable the option
2. Possibility to limit the number of options that can be selected
3. Add image support to the option

### Demonstration:

msedge_HiaAJSDWNm

### CODE REVIEW

File: src/View/Components/Select/Styled.php [hunk 1]
```diff
@@ -92,6 +93,8 @@ public function personalization(): array
                         'wrapper' => 'dark:text-dark-300 dark:hover:bg-dark-500 dark:focus:bg-dark-500 relative cursor-pointer select-none px-2 py-2 text-gray-700 transition hover:bg-gray-100 focus:bg-gray-100 focus:outline-none',
                         'options' => 'flex items-center justify-between',
                         'selected' => 'font-semibold hover:bg-red-500 hover:text-white dark:hover:bg-red-500',
+                        'disabled' => 'dark:bg-dark-500 !cursor-not-allowed bg-gray-100',
+                        'img' => 'h-6 w-6 rounded-full',
```
Comment (devajmeireles) [inline src/View/Components/Select/Styled.php]:
```suggestion
                        'image' => 'h-6 w-6 rounded-full',
```

File: src/View/Components/Select/Traits/InteractsWithSelectOptions.php [hunk 2]
```diff
@@ -14,7 +14,12 @@ private function options(): void
         $label = explode(':', $select[0])[1];
         $value = explode(':', $select[1])[1];
 
-        $this->options = collect($this->options)->map(fn (array $item) => [$label => $item[$label], $value => $item[$value]])->toArray();
+        $this->options = collect($this->options)->map(fn (array $item) => [
+            $label => $item[$label],
+            $value => $item[$value],
+            'disabled' => $item['disabled'] ?? false,
+            'img' => current(array_intersect_key($item, array_flip(['image', 'img', 'preview', 'img_src']))) ?: null,
```
Comment (devajmeireles) [inline src/View/Components/Select/Traits/InteractsWithSelectOptions.php]:
```suggestion
            'image' => current(array_intersect_key($item, array_flip(['image', 'img', 'preview', 'img_src']))) ?: null,
```

File: tests/Browser/Select/StyledCommonTest.php [hunk 3]
```diff
@@ -291,6 +291,49 @@ public function can_select_multiple_with_same_label(): void
             ->waitForText('baz');
     }
 
+    /** @test */
+    public function can_select_with_disabled_option(): void
```
Comment (devajmeireles) [inline tests/Browser/Select/StyledCommonTest.php]: Which of these tests validates that it is not possible to select a disabled option?

**Review by devajmeireles — CHANGES_REQUESTED** [review]

File: tests/Browser/Select/StyledCommonTest.php (hunk 3 above)
Comment (hitaloramon) [reply tests/Browser/Select/StyledCommonTest.php]: can_select_with_disabled_option

**Review by devajmeireles — APPROVED** [review]

**Comment by devajmeireles** [comment]
Thanks, @hitaloramon !

I just removed the `preview` and formatted the JS.

**Pull request merged**